test/pthread/join_return_fail.c

```diff
@@ -0,0 +1,37 @@
+// Tests join & return in general - would fail if join doesn't block.
+//  Also tests that exit and join successfully store/load a return value
```

Comment:
Hmmm, again, why these extra spaces in the beginning of lines? Please align those properly.